novawallet/Modules/Gifts/Common/RemoteBalanceQueryFacade.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import Operation_iOS
+import BigInt
+
+protocol RemoteBalanceQueryFacadeProtocol {
+    func createTransferrableWrapper(
+        for accountId: AccountId,
+        chainAsset: ChainAsset
+    ) -> CompoundOperationWrapper<BigUInt>
+}
+
+final class RemoteBalanceQueryFacade {
+    private let chainRegistry: ChainRegistryProtocol
+    private let operationQueue: OperationQueue
+
+    init(
+        chainRegistry: ChainRegistryProtocol,
+        operationQueue: OperationQueue
+    ) {
+        self.chainRegistry = chainRegistry
+        self.operationQueue = operationQueue
+    }
+}
+
+// MARK: - Private
+
+private extension RemoteBalanceQueryFacade {
+    func querySubstrateTransferrableBalance(
+        for accountId: AccountId,
+        chainAsset: ChainAsset
+    ) -> CompoundOperationWrapper<BigUInt> {
+        let wrapper = querySubstrateBalance(
+            for: accountId,
+            chainAsset: chainAsset
+        )
+
+        let mapOperation = ClosureOperation {
+            try wrapper.targetOperation
+                .extractNoCancellableResultData()
+                .transferable
+        }
+
+        mapOperation.addDependency(wrapper.targetOperation)
+
+        return wrapper.insertingTail(operation: mapOperation)
+    }
+
+    func querySubstrateBalance(
+        for accountId: AccountId,
+        chainAsset: ChainAsset
+    ) -> CompoundOperationWrapper<AssetBalance> {
+        WalletRemoteQueryWrapperFactory(
+            chainRegistry: chainRegistry,
+            operationQueue: operationQueue
+        ).queryBalance(
+            for: accountId,
+            chainAsset: chainAsset
+        )
+    }
+
+    func queryEthereumBalance(
+        for accountId: AccountId,
+        chainAsset: ChainAsset
+    ) -> CompoundOperationWrapper<BigUInt> {
+        do {
+            let address = try accountId.toAddress(using: chainAsset.chain.chainFormat)
+
+            return EvmRemoteBalanceQueryFactory(
```

Comment:
Why do we need one more facade. I imagine that we could add support for evm tokens to the `WalletRemoteQueryWrapperFactory`. WDYT?